File: api/src/main/java/com/example/warehouse/service/contract/ShipperService.java
package com.example.warehouse.service.contract;

import com.example.warehouse.entity.Shipper;
import com.example.warehouse.entity.ShipperType;
import java.util.List;

public interface ShipperService {
    Shipper createShipper(Shipper shipper);

    Shipper getShipperById(String shipperId);

    List<Shipper> getAllShippers();

    List<Shipper> getShippersByType(ShipperType type);

    List<Shipper> getActiveShippers();

    Shipper updateShipper(String shipperId, Shipper shipper);

    void deleteShipper(String shipperId);

    Shipper restoreShipper(String shipperId);
}

File: web/src/services/ShipperService.ts
import httpClient from '@/lib/httpClient';
import type { ShipperResponse, ShipperRequest } from '@/types/api';

export const ShipperService = {
    getAllShippers: (): Promise<ShipperResponse[]> =>
        httpClient.get('/shippers'),

    getActiveShippers: (): Promise<ShipperResponse[]> =>
        httpClient.get('/shippers/active'),

    createShipper: (shipper: ShipperRequest): Promise<ShipperResponse> =>
        httpClient.post('/shippers', shipper),

    updateShipper: (
        shipperId: string,
        shipper: Partial<ShipperRequest>,
    ): Promise<ShipperResponse> =>
        httpClient.put(`/shippers/${shipperId}`, shipper),

    deleteShipper: (shipperId: string): Promise<void> =>
        httpClient.delete(`/shippers/${shipperId}`),
};
